HoloLensCommander/HoloLensCommander/Dialogs/GetAppInstallFilesDialogViewModelCommands.cs

```diff
@@ -96,6 +104,60 @@ public ICommand RemoveDependencyFileCommand
         }
 
         /// <summary>
+        /// Implementation of the browse for parents folder command.
+        /// </summary>
+        /// <returns>Task object used for tracking method completion.</returns>
+        private async Task BrowseForParentFolderAsync()
+        {
+            FolderPicker folderPicker = new FolderPicker();
+            folderPicker.ViewMode = PickerViewMode.List;
+            folderPicker.SuggestedStartLocation = PickerLocationId.ComputerFolder;
+            folderPicker.CommitButtonText = "Select";
+            folderPicker.FileTypeFilter.Add("*");
+
+            StorageFolder folder = await folderPicker.PickSingleFolderAsync();
+            if (folder!=null)
+            {
+                string appxFolderToken = StorageApplicationPermissions.FutureAccessList.Add(folder);
+                await AutoAddInstallFiles(appxFolderToken);
+            }
+        }
+
+        private async Task AutoAddInstallFiles(string appxFolderToken)
+        {
+            StorageFolder appPackageFolder = await StorageApplicationPermissions.FutureAccessList.GetFolderAsync(appxFolderToken);
+            IReadOnlyList<StorageFile> files = await appPackageFolder.GetFilesAsync();
+
+            Func<StorageFile, bool> appxFilter = f => f.FileType.Equals(".appx") || f.FileType.Equals(".appxbundle");
+            StorageFile appxFile = files.FirstOrDefault(appxFilter);
+            if (appxFile == null)
+            {
+                return;
+            }
+
+            this.AppPackageFile = appxFile;
+
+            StorageFile certFile = files.FirstOrDefault(f => f.FileType.Equals(".cer"));
+            if (certFile == null)
+            {
+                return;
+            }
+
+            this.AppCertificateFile = certFile;
+
+            DependencyFiles.Clear();
+            DependencyFileNames.Clear();
+
+            StorageFolder dependenciesFolder = await appPackageFolder.GetFolderAsync(@"Dependencies\x86");
```

Comment:
Yea, was thinking of radio buttons in the meantime. Does this change to the device portal wrapper you're working on allow for easy detection of the type of device that's connected? And does the device type always dictate a specific dependency architecture?